### PR TITLE
feat: WDS algorithmic negative colors

### LINKED ISSUE #22365: feat: WDS negative color tokens
Currently: hardcoded values.

Needed: check the seed for `isRed`, if it is: pick a maximally distant hue in the Red range (`29..50`), otherwise pick some default. Contrast and other checks for `-accent-` colors apply with minor adjustments (`negative`, being an error state color, can/should be more prominent that `accent`).

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Checking whether seed falls into the interval and picking most distant red
Fixes #22365   

## Media

https://github.com/appsmithorg/appsmith/assets/80973/4dc4d9bb-1baa-4d34-87d0-57f08b7af2c2

## Type of change

- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts [hunk 1]
```diff
@@ -199,6 +228,43 @@ export class DarkModeTheme implements ColorModeTheme {
...
+
+    if (this.seedLightness < 0.4) {
+      color.oklch.l = 0.4;
+    }
+
+    return color;
+  }
+
+  private get bdNegative() {
+    const color = this.bdAccent.clone();
```
Comment (ichik) [inline app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts:244]: This will fail with an achromatic seed (`bdAccent` has no chroma).
Color needs to be created from first principles.

File: app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts [hunk 2]
```diff
@@ -120,6 +124,22 @@ export class DarkModeTheme implements ColorModeTheme {
     return this.fg.clone();
   }
 
+  private get bgNegative() {
+    const color = this.bgAccent.clone();
```
Comment (ichik) [inline app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts:128]: Similarly zero chroma `bgAccent` is not taken into consideration here.

File: app/client/packages/design-system/theming/src/color/ColorsAccessor.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ export class ColorsAccessor {
   }
 
   get isRed() {
-    return this.color.oklch.h >= 29 && this.color.oklch.h <= 50;
+    return this.color.oklch.h >= 5 && this.color.oklch.h <= 49;
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/color/ColorsAccessor.ts:50]: @ichik (just thoughts) Maybe we should check other ranges as well.

**Review by KelvinOm — APPROVED** [review]
